4.interpret_model/scripts/nbconverted/interpret_model_coefficients.py

```diff
@@ -15,216 +16,113 @@
-
-
-coefs = np.abs(shuffled_baseline_log_reg_model.coef_)
-coefs = pd.DataFrame(coefs).T
-coefs.columns = shuffled_baseline_log_reg_model.classes_
-
-print(coefs.shape)
-coefs.head()
-
-
-# ### Diagrams for interpreting coefficients
-
-# In[14]:
-
-
-# display heatmap of average coefs
-plt.figure(figsize=(20, 10))
-plt.title("Heatmap of Coefficients Matrix")
-ax = sns.heatmap(data=coefs.T)
-
-
-# In[15]:
-
-
-# display clustered heatmap of coefficients
-ax = sns.clustermap(data=coefs.T, figsize=(20, 10), row_cluster=True, col_cluster=True)
-ax = ax.fig.suptitle("Clustered Heatmap of Coefficients Matrix")
-
-
-# In[16]:
-
-
-# display density plot for coefficient values of each class
-sns.set(rc={"figure.figsize": (20, 8)})
-plt.xlim(-0.02, 0.1)
-plt.xlabel("Coefficient Value")
-plt.ylabel("Density")
-plt.title("Density of Coefficient Values Per Phenotpyic Class")
-ax = sns.kdeplot(data=coefs)
+# ### Load Labeled Data
+# 
 
+# In[2]:
 
-# In[17]:
 
+# load labeled data
+labeled_data_path = pathlib.Path("../0.download_data/data/labeled_data.csv.gz")
+labeled_data = get_features_data(labeled_data_path)
+labeled_data
 
-# display average coefficient value vs phenotypic class bar chart
-pheno_class_ordered = coefs.reindex(
-    coefs.mean().sort_values(ascending=False).index, axis=1
-)
-sns.set(rc={"figure.figsize": (20, 8)})
-plt.xlabel("Phenotypic Class")
-plt.ylabel("Average Coefficient Value")
-plt.title("Coefficient vs Phenotpyic Class")
-plt.xticks(rotation=90)
-ax = sns.barplot(data=pheno_class_ordered)
 
+# ### Save Coefficients and Create Interpretation Figures
+# 
 
-# In[18]:
+# In[3]:
 
 
-# display average coefficient value vs feature bar chart
-feature_ordered = coefs.T.reindex(
-    coefs.T.mean().sort_values(ascending=False).index, axis=1
-)
-sns.set(rc={"figure.figsize": (500, 8)})
-plt.xlabel("Deep Learning Feature Number")
-plt.ylabel("Average Coefficient Value")
-plt.title("Coefficient vs Feature")
-plt.xticks(rotation=90)
-ax = sns.barplot(data=feature_ordered)
+# directory to save models to
+coefs_save_directory = pathlib.Path(f"coefficients/")
+coefs_save_directory.mkdir(parents=True, exist_ok=True)
+
+# directory to load the models from
+models_dir = pathlib.Path("../2.train_model/models/")
+
+# use a list to keep track of scores in tidy long format for each model and dataset combination
+compiled_class_PR_curves = []
+
+# iterate through each model (final model, shuffled baseline model, etc)
+# sorted so final models are shown before shuffled_baseline
+for model_path in sorted(models_dir.iterdir()):
+    # load model
+    model = load(model_path)
+    # determine model/feature type from model file name
+    model_type = model_path.name.split("__")[0]
+    feature_type = model_path.name.split("__")[1].replace(".joblib", "")
+
+    print(f"Interpreting model: {model_type} \nTrained with features: {feature_type}")
+
+    # get model coefficients and reshape them into a more useable format
+    coefs = np.abs(model.coef_)
```

Comment:
The direction is important, please add the raw coefficients to the table. You can also separately add the absolute value coefficients as well, but in a seperate column

Reply:
I ended up going with just the raw coefficients as the absolute value can always be determined later. Added in https://github.com/WayScience/phenotypic_profiling_model/pull/22/commits/d00f94dbaa9c5d36fc43ee0af5432ba72dba46bd